test/suites/edgexfoundry/main_test.go

```diff
@@ -16,37 +16,47 @@ const (
 	supportSchedulerServicePort = "59861"
 )
 
-func TestMain(m *testing.M) {
-	// start clean
+func main(m *testing.M) (int, error) {
+	log.Println("[CLEAN]")
 	utils.SnapRemove(nil,
 		platformSnap,
 	)
 
 	log.Println("[SETUP]")
+
 	start := time.Now()
+	defer utils.SnapDumpLogs(nil, start, platformSnap)
+
+	var err error
 
 	if utils.LocalSnap != "" {
-		utils.SnapInstallFromFile(nil, utils.LocalSnap)
+		err = utils.SnapInstallFromFile(nil, utils.LocalSnap)
 	} else {
-		utils.SnapInstallFromStore(nil, platformSnap, utils.ServiceChannel)
+		err = utils.SnapInstallFromStore(nil, platformSnap, utils.ServiceChannel)
+	}
+	if err != nil {
+		return 0, err
 	}
+	defer utils.SnapRemove(nil, platformSnap)
```

Comment:
Remove after logging so that shutdown logs aren't included. This can be done by moving the deferred removal to before deferred logging.